File: PYG-SellerGoods-Interface/src/main/java/it/com/pyg/sellergoods/service/BrandService.java
package it.com.pyg.sellergoods.service;

import java.util.List;

import it.com.pyg.bean.PageResult;
import it.com.pyg.bean.Result;
import it.com.pyg.entity.Brand;


public interface BrandService {

	//获取所有
	List<Brand> getAll();
	
	//分页获取
	PageResult getByPage(Integer page,Integer size);

	//根据id获取
	Result getById(Long id);

	//根据id删除
	Result delete(Long id);

	//批量删除
	Result deleteBatch(Long[] ids);

	//新增
	Result add(Brand brand);

	//修改
	Result update(Brand brand);

	//分页条件查询
	PageResult search(Brand brand,Integer page,Integer size);
}

File: PYG-Manager-Web/src/main/webapp/js/service/brandService.js
//品牌服务层
app.service('brandService',function($http){
	//分页获取
	this.getByPage = function(page,rows){
		return $http.get('../brand/getByPage.do?page=' + page + '&size=' + size);
	}
	//查询实体
    this.getById = function(id){
        return $http.get('../brand/get/' + id + ".do");
    }
	//新增
    this.add = function(entity){
        return $http.post('../brand/add.do',entity);
    }
	//修改
    this.update = function(entity){
        return $http.post('../brand/update.do',entity);
    }
	//删除
    this.dele = function(ids){
        return $http.get('../brand/deleteBatch.do?ids=' + ids);
    }
	//搜索
	this.search = function(page,size,searchEntity){
		return $http.post('../brand/search.do?page='+page+"&size="+size, searchEntity);
	}  
});